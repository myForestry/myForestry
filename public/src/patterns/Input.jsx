import React, { Component } from 'react';

export default class Input extends Component {
  static propTypes = {
    initialValue: React.PropTypes.string,
    label: React.PropTypes.string,
    name: React.PropTypes.string,
    onChange: React.PropTypes.func,
    placeholder: React.PropTypes.string,
  }

  state = {
    value: '',
  };

  componentWillMount() {
    if (this.props.initialValue) this.setState({ value: this.props.initialValue });
  }

  _handleChange = (event) => {
    this.setState({ value: event.target.value });

   if (this.props.onChange && typeof (this.props.onChange === 'function')) this.props.onChange(event.target.name, event.target.value);
  }

  render() {
    const { label, placeholder, name } = this.props;

    return (
      <div className="form-group">
        <label htmlFor={label}>{label}</label>
        <input className="form-control" name={name} onChange={this._handleChange} type="text" value={this.state.value} placeholder={placeholder} />
      </div>
    );
  }
}
