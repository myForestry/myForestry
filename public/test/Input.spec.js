import React from 'react';
import Input from '../src/patterns/Input';
import renderer from 'react-test-renderer';

test('Input has prop name', () => {
  const component = renderer.create(
    <Input name="foo" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.children[1].props.name).toBe('foo');
});

test('Input has className form-group', () => {
  const component = renderer.create(
    <Input />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.props.className).toBe('form-group');
});

test('Input has children', () => {
  const component = renderer.create(
    <Input />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.children).toHaveLength(2);

  for (var i = 0; i < tree.children.length; i++) {
    expect(tree.children[i].children).toBe(null);
  }
});

test('Input accepts a label', () => {
  const component = renderer.create(
    <Input label="foo" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.children[0].children).toContain('foo');
});

test('Input accepts an initial value', () => {
  const component = renderer.create(
    <Input initialValue="foo" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.children[1].props.value).toBe('foo');
});
