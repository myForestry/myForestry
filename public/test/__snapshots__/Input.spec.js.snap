exports[`test Input accepts a label 1`] = `
<div
  className="form-group">
  <label
    htmlFor="foo">
    foo
  </label>
  <input
    className="form-control"
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
</div>
`;

exports[`test Input accepts an initial value 1`] = `
<div
  className="form-group">
  <label
    htmlFor={undefined} />
  <input
    className="form-control"
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="foo" />
</div>
`;

exports[`test Input has children 1`] = `
<div
  className="form-group">
  <label
    htmlFor={undefined} />
  <input
    className="form-control"
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
</div>
`;

exports[`test Input has className form-group 1`] = `
<div
  className="form-group">
  <label
    htmlFor={undefined} />
  <input
    className="form-control"
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
</div>
`;

exports[`test Input has prop name 1`] = `
<div
  className="form-group">
  <label
    htmlFor={undefined} />
  <input
    className="form-control"
    name="foo"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
</div>
`;
